### PR TITLE
fix: add missing params to tekton pipelinerun 'finally'

### PR DESCRIPTION
**Pull request by joshuasimon-taulia**
https://kubernetes.slack.com/archives/C9MBGQJRH/p1666717135265409
i can add tests

this only handles populating `finally[*].params`, `finally[*].taskSpec.params`, and `finally[*].taskSpec.stepTemplate.env`.
the ability to inherit pipelinerun `pipelineSpec.finally` via `uses:` should be added in another PR

### CODE REVIEW

**Comment by ankitm123** [comment]
> i can add tests

Yes, that would be nice!

**Comment by ankitm123** [comment]
/hold
we will remove the hold after tests..

**Comment by joshuasimon-taulia** [comment]
added some

**Comment by ankitm123** [comment]
For completeness, add a test case here: https://github.com/jenkins-x/lighthouse/blob/main/pkg/triggerconfig/inrepo/default_parameters_test.go

File: pkg/triggerconfig/inrepo/test_data/load_pipelinerun/pipeline-finally-params/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+## Reuse steps in a task inherit a 'finally' block
+
+This example shows how we can use `image: uses:sourceURI` and a `name: mystep` to include individual the steps in task
```
Comment (JordanGoasdoue) [inline pkg/triggerconfig/inrepo/test_data/load_pipelinerun/pipeline-finally-params/README.md:3]: I guess this README should be about the params and not the `image: uses:sourceURI` which should be on another PR as you said

Comment (joshuasimon-taulia) [reply pkg/triggerconfig/inrepo/test_data/load_pipelinerun/pipeline-finally-params/README.md]: done

**Comment by joshuasimon-taulia** [comment]
> For completeness, add a test case here: https://github.com/jenkins-x/lighthouse/blob/main/pkg/triggerconfig/inrepo/default_parameters_test.go

done

**Review by ankitm123 — APPROVED** [review]
